components/draw/GeometryMode.svelte

```diff
@@ -0,0 +1,156 @@
+      gjScheme.update((gj) => {
+        let feature = gj.features.find((f) => f.id == currentlyEditing);
+        // Keep the ID and any properties. Just copy over stuff from routeSnapper.
+        // TODO We're depending on implementation details here and knowing what to copy...
+        feature.properties.length_meters = editedRoute.properties.length_meters;
+        feature.properties.waypoints = editedRoute.properties.waypoints;
+        delete feature.properties.hide_while_editing;
+        feature.geometry = editedRoute.geometry;
+        return gj;
+      });
+
+      // Stay in this mode
+      currentlyEditing = null;
+    }
+  });
+
+  $map.on("draw.update", (e) => {
+    if (mode == "edit-geometry") {
+      // Assume there's exactly 1 feature
+      const feature = e.features[0];
+
+      gjScheme.update((gj) => {
+        let updateFeature = gj.features.find((f) => f.id == currentlyEditing);
+        updateFeature.geometry = feature.geometry;
+        return gj;
+      });
+
+      // Stay in this mode
+    }
+  });
+
+  $map.on("draw.selectionchange", (e) => {
+    if (mode == "edit-geometry" && e.features.length == 0) {
+      drawControls.changeMode("static");
+      gjScheme.update((gj) => {
+        let feature = gj.features.find((f) => f.id == currentlyEditing);
+        delete feature.properties.hide_while_editing;
+        return gj;
+      });
+
+      // Stay in this mode
+      currentlyEditing = null;
+    }
+  });
+
+  function startEditing(id) {
+    currentHover.set(null);
+
+    let feature;
+    gjScheme.update((gj) => {
+      feature = gj.features.find((f) => f.id == id);
+      // Hide it from the regular drawing while we're editing
+      feature.properties.hide_while_editing = true;
+      return gj;
+    });
+
+    currentlyEditing = id;
+
+    if (feature.geometry.type == "LineString") {
+      // TODO Oops, we need to fix upstream. If they're missing, we can at
+      // least attempt the endpoints.
+      if (!feature.properties.waypoints) {
+        feature.properties.waypoints = [
+          {
+            lon: feature.geometry.coordinates[0][0],
+            lat: feature.geometry.coordinates[0][1],
+            snapped: true,
+          },
+          {
+            lon: feature.geometry.coordinates[
+              feature.geometry.coordinates.length - 1
+            ][0],
+            lat: feature.geometry.coordinates[
+              feature.geometry.coordinates.length - 1
+            ][1],
+            snapped: true,
+          },
+        ];
+      }
+      routeSnapper.editExisting(feature);
+    } else if (feature.geometry.type == "Polygon") {
+      // The feature already has an ID set, so no need to check what add() returns
+      drawControls.add(feature);
+      drawControls.changeMode("direct_select", { featureId: currentlyEditing });
+    } else if (feature.geometry.type == "Point") {
+      drawControls.add(feature);
+      // We can't direct_select a point
```

Comment:
This is clunky. "Edit geometry" mode, then click an existing point (with the intention of moving it). You then have to click (again) and drag it

Reply:
This is again annoying but don't think it's a blocker at this stage. The UX is annoying with that teeny-tiny little point it shows once selected but I think overriding the style to make the point larger is probably a good way to go.

Reply:
I've got an in-progress followup to replace point and polygon drawing and improve this. For now, agree, it's not a blocker